### ISSUE TITLE
Caja Window Size Determines Whether You Can Rename a File or Not

### ISSUE THREAD

**Issue by machineghost** [mate-desktop/caja#1444]
#### Expected behaviour
I can rename a file, whatever the size of the Caja window

#### Actual behaviour

I try to rename a file with F2: nothing happens.  I go to the menu and do rename: nothing happens.  I right-click and rename: nothing hapoens.

I increase the width of the window ... suddenly it magically works

#### Steps to reproduce the behaviour

Basically just the above, but this thread has further details (NOTE: I didn't post in that thread, but it seems like the same core problem, and it helped me find the resize workaround):

https://forums.linuxmint.com/viewtopic.php?t=296934

#### MATE general version

I literally don't know how to check this: maybe leave a comment here saying what command to run?

#### Package version

See above.

#### Linux Distribution

Linux MInt (not sure exactly but just installed it a month ago)

#### Link to downstream report of your Distribution

See above: you really would get better info if you told people *how* to give you what you want.

**Comment by phisigma** [comment]
First of all, machineghost, _thank you_ very much!

I finally have my workaround 😅
I would never have thought of the window size. I use Mate/Caja 1.24.0 on Ubuntu 20.04.

Just a small hint how to reproduce this issue so you don’t have to read the forum thread:
Create a folder with some subfolders or files, or use an existing folder (with about 5 to 15 objects). Then resize the window so that a vertical scroll bar _just_ appears on the right. This scrollbar could be omitted and “does not really make sense” as you can only scroll 2 or 5 pixels, so it uses nearly the whole space.
Now you cannot rename objects anymore.

**Comment by usb21** [comment]
Just installed Mint 20.1 64bit MATE w/ Caja File Mgr.
I CANNOT rename any files while in full screen, unless i right-click , then go down to "Properties", and then change the name in the "Name" box.
I CAN rename files, either using F2 or right-click and "Rename", when NOT in full screen.

**Comment by Amis** [comment]
Link to downstream:
https://bugs.launchpad.net/bugs/1771768

**Comment by lukefromdc** [comment]
I too have seen this, it can be fiddly at times though usually OK on my system.

**Comment by thesquash** [comment]
I might be wrong, but this sure smells like yet **another** GTK+ bug.

This issue appears in Icon View and Compact View modes; it does not happen in List View mode.  Since both Compact and Icon View modes use the GtkIconView widget (and List View mode uses GtkTreeView instead), the GtkIconView seems to be the common factor.

In all of the below screengrabs, I'm enthusiastically pressing the <kbd>F2</kbd> key in an attempt to rename my `.profile` file.  Only in List View mode do I get anywhere.

![compact_view](https://user-images.githubusercontent.com/66928007/147681948-cfc91df5-1c68-4120-ae8c-9a585a22d465.png)
![icon_view](https://user-images.githubusercontent.com/66928007/147681972-a95dc595-dbca-4f0f-806a-1207da716aa4.png)
![list_view](https://user-images.githubusercontent.com/66928007/147681977-b346bd4d-08c9-4d28-a9e0-8ded5a7d7f20.png)

**Comment by lukefromdc** [comment]
I'm pretty sure the only place we use Gtk's own icon view widget is in mate-appearance-properties, showing the theme thumbnails.

While it is indeed possible to use GtkIconView for a file manager's icon display, Nautilus always used a custom widget, so Nemo and Caja also do. I am guessing the icon view widget in GTK was based originally on a simplification of this, it is known some other widgets first used in Nautilus found their way into GTK 2.

**Comment by thesquash** [comment]
You're right, @lukefromdc -- I started digging into the code and remembered that, sorry.

Anyway, I've made another interesting discovery.  I can reproduce the odd behavior in Icon View only when both of the following are true:

1. The vertical scrollbar is showing (i.e., the FMIconContainer is too large to fit entirely inside the window); and
2. The FMIconContainer is allocated an area **less than** 16 pixels **shorter** than its minimum height (i.e., you can scroll the page by 15 pixels or less).

The 16 pixel "magic number" is not affected by the width / height of the scrollbar, as I tried this with themes with 13, 14, 15, and even 4 pixel scrollbars and the 16 pixel magic number did not change.

**Comment by lukefromdc** [comment]
I used mate-search-tool to look for the number or string "16" in all the files in caja's src directory and found nothing. Thus this is not a magic number set somewhere in our code with unexpected results

**Comment by halfburnttoast** [comment]
I'm having this issue too with v1.26.0 shipped with Linux Mint 21. Seems like almost randomly, file renaming stops working. But, if I slightly resize the window, renaming works again.
